profiles/common/modules/features/nexteuropa_newsroom/theme/newsroom-agenda.tpl.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * @file
+ * Agenda.
+ */
+?>
+<?php if (!empty($filter_form)): ?>
+<div class="newsroomAgenda-filterForm">
+  <h3><?php echo t('filter by');?></h3>
+  <?php echo drupal_render($filter_form); ?>
+</div>
+<?php endif; ?>
+<div class="<?php echo !$is_block ? 'newsroomAgenda-container' : NULL; ?>">
+  <div class="currentDate">
+    <?php if (count($items) > 0) : ?>
+      <?php foreach ($items as $agenda_item): ?>
+        <div class="date-row">
+          <div class="date-date">
+            <span class="day"><?php echo $agenda_item->getDate()->format('d'); ?></span>
+            <span class="month"><?php echo drupal_strtoupper($agenda_item->getDate()->format('M')); ?></span>
+            <span class="year"><?php echo $agenda_item->getDate()->format('Y'); ?></span>
+          </div>
+          <div class="date-events">
+            <?php if (count($agenda_item->getItems()) > 0) : ?>
```

Comment:
Please put the whole rendering from this statement to a dedicated template function and make it less mixed with logic here.